src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java

```diff
@@ -0,0 +1,410 @@
+import uk.gov.hmcts.darts.arm.model.record.UploadNewFileRecord;
+import uk.gov.hmcts.darts.arm.model.record.armresponse.ArmResponseUploadFileRecord;
+import uk.gov.hmcts.darts.arm.service.ArmResponseFilesProcessor;
+import uk.gov.hmcts.darts.arm.util.files.InputUploadFilenameProcessor;
+import uk.gov.hmcts.darts.arm.util.files.UploadFileFilenameProcessor;
+import uk.gov.hmcts.darts.authorisation.component.UserIdentity;
+import uk.gov.hmcts.darts.common.entity.AnnotationDocumentEntity;
+import uk.gov.hmcts.darts.common.entity.ExternalLocationTypeEntity;
+import uk.gov.hmcts.darts.common.entity.ExternalObjectDirectoryEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectRecordStatusEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
+import uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum;
+import uk.gov.hmcts.darts.common.repository.ExternalLocationTypeRepository;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectRecordStatusRepository;
+import uk.gov.hmcts.darts.common.service.FileOperationService;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.nonNull;
+import static uk.gov.hmcts.darts.arm.util.ArchiveConstants.ArchiveRecordOperationValues.ARM_FILENAME_SEPARATOR;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_DROP_ZONE;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_PROCESSING_RESPONSE_FILES;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_PROCESSING_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ArmResponseFilesProcessorImpl implements ArmResponseFilesProcessor {
+
+    public static final String ARM_RESPONSE_FILE_EXTENSION = ".rsp";
+    public static final String ARM_INPUT_UPLOAD_FILENAME_KEY = "iu";
+    public static final String ARM_CREATE_RECORD_FILENAME_KEY = "cr";
+    public static final String ARM_UPLOAD_FILE_FILENAME_KEY = "uf";
+    public static final String ARM_RESPONSE_SUCCESS_STATUS_CODE = "1";
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final ArmDataManagementApi armDataManagementApi;
+    private final FileOperationService fileOperationService;
+    private final ArmDataManagementConfiguration armDataManagementConfiguration;
+    private final ObjectMapper objectMapper;
+    private final UserIdentity userIdentity;
+
+    @Transactional
+    @Override
+    public void processResponseFiles() {
```

Comment:
is this separate method required?

Reply:
I think I just followed what the other tasks were doing. Removed